pkg/registry/apis/provisioning/jobs/export/resources.go

```diff
@@ -14,6 +16,8 @@ import (
 	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
 )
 
+type versionUpgrader = func(ctx context.Context, item *unstructured.Unstructured) (*unstructured.Unstructured, error)
```

Comment:
I think it's better to save this pattern for other occasion in which we have a separate component to determine strategy of when the logic is not anymore within the worker. 

I think a simple method `maintainDashboardOriginalVersionIfConversionFailed` would be easier to follow. 

The other option and probably better is to extract out this logic into something like a "conversion fixer" or similar. 

Other option would be to wrap our dashboards client to do this transparently when listing or getting dashboards.

Reply:
🤷🏻  no strong feeling other than this pattern seems flawed for our long term strategy -- for sure not something we want users to configure.  Ideally we can get the saved version directly and not iterate conversions for each value.

I don't really want this patten to expand too much -- it is a bandaid that will need to be fixed along with getting history from unified storage also